apps/designer/app/routes/$.tsx

```diff
@@ -0,0 +1,128 @@
+import { type LoaderFunction, redirect, MetaFunction } from "@remix-run/node";
+import { useLoaderData } from "@remix-run/react";
+import { InstanceRoot, Root } from "@webstudio-is/react-sdk";
+import { loadCanvasData, type CanvasData } from "~/shared/db";
+import config from "~/config";
+import { db } from "@webstudio-is/project/index.server";
+import env, { Env } from "~/env.server";
+import { sentryException } from "~/shared/sentry";
+import { Canvas } from "~/canvas";
+
+type Mode = "edit" | "preview" | "published";
+
+type Data =
+  | (CanvasData & { env: Env; mode: Mode })
+  | { errors: string; env: Env };
+
+type ProjectIdentifier = { type: "id" | "domain"; value: string };
+
+export const isCanvasRequest = (
```

Comment:
We should add tests for this at this stage, if pass request.url and request.headers, it will be easy to test

Reply:
Tests would be more important right now just to see all the cases it handles

Reply:
isCanvasRequest sounds like a function that returns booleans only, but since it can return object, it needs a better name

Reply:
maybe call it parseUrl or something and return undefined when can't parse?

Reply:
- added tests
- renamed to `getCanvasRequestParams`
- returns `undefined` if doesn't match